Apps/Encounter/src/Delegates/RestMSPVisitDelegate.cs

```diff
@@ -0,0 +1,161 @@
+        private readonly ITraceService traceService;
+        private readonly IHttpClientService httpClientService;
+        private readonly IConfiguration configuration;
+        private readonly ODRConfig odrConfig;
+        private readonly Uri baseURL;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RestMSPVisitDelegate"/> class.
+        /// </summary>
+        /// <param name="logger">Injected Logger Provider.</param>
+        /// <param name="traceService">Injected TraceService Provider.</param>
+        /// <param name="httpClientService">The injected http client service.</param>
+        /// <param name="configuration">The injected configuration provider.</param>
+        public RestMSPVisitDelegate(
+            ILogger<RestMSPVisitDelegate> logger,
+            ITraceService traceService,
+            IHttpClientService httpClientService,
+            IConfiguration configuration)
+        {
+            this.logger = logger;
+            this.traceService = traceService;
+            this.httpClientService = httpClientService;
+            this.configuration = configuration;
+            this.odrConfig = new ODRConfig();
+            this.configuration.Bind(ODRConfigSectionKey, this.odrConfig);
+            if (this.odrConfig.DynamicServiceLookup)
+            {
+                string? serviceHost = Environment.GetEnvironmentVariable($"{this.odrConfig.ServiceName}{this.odrConfig.ServiceHostSuffix}");
+                string? servicePort = Environment.GetEnvironmentVariable($"{this.odrConfig.ServiceName}{this.odrConfig.ServicePortSuffix}");
+                Dictionary<string, string> replacementData = new Dictionary<string, string>()
+                {
+                    { "serviceHost", serviceHost! },
+                    { "servicePort", servicePort! },
+                };
+                this.baseURL = new Uri(StringManipulator.Replace(this.odrConfig.BaseEndpoint, replacementData) !);
+            }
+            else
+            {
+                this.baseURL = new Uri(this.odrConfig.BaseEndpoint);
+            }
+
+            logger.LogInformation($"ODR Proxy URL resolved as {this.baseURL.ToString()}");
+        }
+
+        /// <inheritdoc/>
+        public async Task<RequestResult<MSPVisitHistoryResponse>> GetMSPVisitHistoryAsync(ODRHistoryQuery query, string hdid, string ipAddress)
+        {
+            using ITracer tracer = this.traceService.TraceMethod(this.GetType().Name);
+            if (query == null)
+            {
+                throw new ArgumentNullException(nameof(query), "Query cannot be null");
+            }
+            else if (query.PHN == null)
+            {
+                throw new ArgumentNullException(nameof(query), "Query PHN cannot be null");
+            }
+
+            RequestResult<MSPVisitHistoryResponse> retVal = new RequestResult<MSPVisitHistoryResponse>();
+            using (this.traceService.TraceSection(this.GetType().Name, "ODRQuery"))
```

Comment:
@sslaws has just updated our code to the latest C# 8 framework, we can simplify this using statement now <https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement>